tajo-plan/src/main/java/org/apache/tajo/plan/rewrite/rules/PartitionedTableRewriter.java

```diff
@@ -118,13 +120,76 @@ public String toString() {
    * @return
    * @throws IOException
    */
-  private Path [] findFilteredPaths(OverridableConf queryContext, Schema partitionColumns, EvalNode [] conjunctiveForms,
-                                    Path tablePath)
-      throws IOException {
+  private Path [] findFilteredPaths(OverridableConf queryContext, String tableName,
+                                    Schema partitionColumns, EvalNode [] conjunctiveForms, Path tablePath)
+      throws IOException, UndefinedDatabaseException, UndefinedTableException, UndefinedPartitionMethodException,
+      UndefinedOperatorException, UnsupportedException {
 
+    Path [] filteredPaths = null;
     FileSystem fs = tablePath.getFileSystem(queryContext.getConf());
+    String [] splits = CatalogUtil.splitFQTableName(tableName);
+    List<PartitionDescProto> partitions = null;
+
+    try {
+      if (conjunctiveForms == null) {
+        partitions = catalog.getPartitionsOfTable(splits[0], splits[1]);
+        if (partitions.isEmpty()) {
+          filteredPaths = findFilteredPathsFromFileSystem(partitionColumns, conjunctiveForms, fs, tablePath);
+        } else {
+          filteredPaths = findFilteredPathsByPartitionDesc(partitions);
+        }
+      } else {
+        if (catalog.existPartitions(splits[0], splits[1])) {
+          PartitionsByAlgebraProto request = getPartitionsAlgebraProto(splits[0], splits[1], conjunctiveForms);
+          partitions = catalog.getPartitionsByAlgebra(request);
+          filteredPaths = findFilteredPathsByPartitionDesc(partitions);
+        } else {
+          filteredPaths = findFilteredPathsFromFileSystem(partitionColumns, conjunctiveForms, fs, tablePath);
+        }
+      }
+    } catch (UnsupportedException ue) {
```

Comment:
IMO, this case is regarded as the full scan on partitions.
So, the same routine with the above ```if (conjunctiveForms == null)``` clause is required here.